File: ios/ExpoBarcodeModule.swift
import ExpoModulesCore

public class ExpoBarcodeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoBarcode")

    Events("onChangeTheme")

    Function("setTheme") { (theme: String) -> Void in
      UserDefaults.standard.set(theme, forKey:"theme")
      sendEvent("onChangeTheme", [
        "theme": theme
      ])
    }

    Function("getTheme") { () -> String in
      UserDefaults.standard.string(forKey: "theme") ?? "system"
    }
  }

  enum Theme: String, Enumerable {
    case light
    case dark
    case system
  }
}

File: android/src/main/java/expo/modules/barcode/ExpoBarcodeModule.kt
package expo.modules.barcode

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoBarcodeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoBarcode")

    Function("getTheme") {
      return@Function "system"
    }
  }
}
